src/bqlfn.py

```diff
@@ -280,6 +281,35 @@ def bql_column_value_probability(bdb, generator_id, modelno, colno, value,
     r = metamodel.logpdf_joint(
         bdb, generator_id, targets, constraints, modelno)
     return ieee_exp(r)
+
+# XXX This is silly.  We should return log densities, not densities.
+def bql_pdf_joint(bdb, generator_id, modelno, *args):
+    metamodel = core.bayesdb_generator_metamodel(bdb, generator_id)
+    # A nonexistent (`unobserved') row id.
+    fake_row_id = core.bayesdb_generator_fresh_row_id(bdb, generator_id)
+    i = 0
+    targets = []
+    while i < len(args):
+        if args[i] == -1:
+            i += 1
+            break
+        if i + 1 == len(args):
+            raise ValueError('Odd logpdf targets/constraints: %r' % (args,))
```

Comment:
Odd here is most naturally read to mean "unusual", but in fact, you mean numerically odd, rather than even. Consider rewording as "Expected targets and their constraints to come as pairs, but found an odd number of targets and constraints"?